[Bug] Failed to build BE on macOS arm64

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.


### Version

Doris Version: master
OS: macOS Ventura 13.1 (arm64)

### What's Wrong?

```shell
Undefined symbols for architecture arm64:
  "_p4nd1enc256v32", referenced from:
      lucene::index::IndexWriter::mergeTerms() in libclucene-core-static.a(IndexWriter.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

### What You Expected?

Build BE successfully.

### How to Reproduce?

```shell
./build.sh --be
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/thirdparty/build-thirdparty.sh
+++ b/thirdparty/build-thirdparty.sh
@@ -1549,15 +1549,21 @@ build_concurrentqueue() {
 
 #clucene
 build_clucene() {
-    if [[ -z ${USE_AVX2} ]]; then
-        USE_AVX2=1
+    if [[ "$(uname -m)" == 'x86_64' ]]; then
+        USE_AVX2="${USE_AVX2:-1}"
+    else
+        USE_AVX2="${USE_AVX2:-0}"
     fi
+
     if [[ -z ${BUILD_TYPE} ]]; then
         BUILD_TYPE=Release
     fi
+
     check_if_source_exist "${CLUCENE_SOURCE}"
     cd "${TP_SOURCE_DIR}/${CLUCENE_SOURCE}"
-    mkdir -p "${BUILD_DIR}" && cd "${BUILD_DIR}"
+
+    mkdir -p "${BUILD_DIR}"
+    cd "${BUILD_DIR}"
     rm -rf CMakeCache.txt CMakeFiles/
 
     ${CMAKE_CMD} -G "${GENERATOR}" -DCMAKE_INSTALL_PREFIX="${TP_INSTALL_DIR}" -DBUILD_STATIC_LIBRARIES=ON \

